### ISSUE TITLE
Investigate options for javac detection on OSX

### ISSUE THREAD

**Issue by peterebden** [thought-machine/please#397]
Looks like in at least some cases, doing `hash javac` works but the `javac` binary is not actually the Java compiler, but something that pops up a window prompting to install it.

See if we can find a way of detecting if it's *really* installed.

**Comment by rozuur** [comment]
@BNMetrics Are you working on this? If not can I work on this?

**Comment by BNMetrics** [comment]
@rozuur No, I'm not currently working on this one at the moment, but feel free to work on this if you have the time! :)

Thank you!

**Comment by rozuur** [comment]
I looked around and found that except for bootstrap.sh javac is not directly used. Is it the only place where the check needs to be corrected?

**Comment by BNMetrics** [comment]
@rozuur Yes! 👍 That would be it.
The `bootstrap.sh` is the only place javac is being referenced. So I think you might want to check the os system to see if it's `darwin`, so perhaps find a way to turn the popup off. :)

Thanks for investing the time into please! :)

**Comment by rozuur** [comment]
I browsed around and found no easy and recommended way to disable popups for javac. The reason this happens is due to a third-party application installing incompatible java[1]. The user needs to install a new version as discussed in the thread. 

1. https://discussions.apple.com/thread/7701084

**Comment by peterebden** [comment]
Thanks for looking into it!

I think what we might end up doing is moving the Java support stuff into separate repos and pull them back via Maven. They aren't directly linked to this repo any more (there used to be some shared proto stuff but that's been gone for a while) and that would avoid this problem - and be a simpler answer about what's required to build plz itself.

There's also a possibility of downloading them dynamically rather than shipping with plz, but I'm not 100% sure that's the right direction, and it'd need a bit more work first.
